util/threadpool_imp.cc

```diff
@@ -320,6 +366,8 @@ void ThreadPoolImpl::Impl::SetBackgroundThreadsInternal(int num,
   if (num > total_threads_limit_ ||
       (num < total_threads_limit_ && allow_reduce)) {
     total_threads_limit_ = std::max(0, num);
+    // Reset the number of reserved threads when total_threads_limit_ changes
+    reserved_threads_.store(0, std::memory_order_relaxed);
```

Comment:
Is it needed? User may call `SetBackgroundThreads()` at any time and RocksDB has no way to know its reservations have been invalidated.

Reply:
Yes. I will remove this.